code/modules/hydroponics/gene_modder.dm

```diff
@@ -24,14 +28,14 @@
 	component_parts += new /obj/item/weapon/stock_parts/console_screen(src)
 	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
 	component_parts += new /obj/item/weapon/stock_parts/micro_laser(src)
-	component_parts += new /obj/item/weapon/stock_parts/micro_laser(src)
 	component_parts += new /obj/item/weapon/stock_parts/manipulator(src)
 	RefreshParts()
 
 /obj/machinery/plantgenes/RefreshParts()
 	rating = 0
 	for(var/obj/item/weapon/stock_parts/S in component_parts)
 		rating += S.rating-1
+	max_extract_pot = initial(max_extract_pot) + rating*5
```

Comment:
No, it maxes out at 95%. See, each parts adds it's (T level - 1) to the device rating. And we have 3 rated parts. Rating is 3 for T2, 9 for T4. It then gets multiplied by 5 and added to 50. Results:
T1: 50 + (1-1)*3*5 = 50
T2: 50 + (2-1)*3*5 = 65
T3: 50 + (3-1)*3*5 = 80
T4: 50 + (4-1)*3*5 = 95